### PR TITLE
typechecker: Infer correct type for negative ints

### LINKED ISSUE #1437: No number type inference for negative numbers
```
fn number_inference() -> i32 {
    return -1
}
```
`Type mismatch: expected ‘i32’, but got ‘i64’`

Would expect this to work as it does for zero or positive numbers and infer i32 type based on the return type.

### PR DESCRIPTION
**Pull request by abooishaaq**
Fixes #1437.

### CODE REVIEW

**Comment by cybergsus** [comment]
Looks great! If you could add a regression test to the test suite, that'd be great as well. Something in the lines of:
```
/// Expected:
/// - output: ""

// brought up by #1437 

fn number_inference() -> i32 {
    return -1
}

fn main() {}
```

**Comment by 0GreenClover0** [comment]
The test fails because it's actually `/// Expect:` instead of `/// Expected:`

File: selfhost/typechecker.jakt [hunk 1]
```diff
@@ -7643,6 +7643,7 @@ struct Typechecker {
         UnaryOp(expr, op, span) => {
             let checked_expr = match op {
                 Dereference => .typecheck_expression(expr, scope_id, safety_mode, type_hint: None)
+                Negate => .typecheck_expression_and_dereference_if_needed(expr, scope_id, safety_mode, type_hint, span)
```
Comment (alimpfard) [inline selfhost/typechecker.jakt:7646]: Please make the commit message explain what's being done, not its result ("typechecker: Pass the correct type hint for unary negate" or something like that)

**Review by alimpfard — COMMENTED** [review]
changes LGTM, just one nitpick:

**Comment by robryanx** [comment]
Looks good! Fixes the issue.

**Pull request merged**